src/components/connection_handler/CMakeLists.txt

```diff
@@ -28,49 +28,38 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
-if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-  find_package(WinOpenSSL REQUIRED)
-endif()
+include(${CMAKE_SOURCE_DIR}/tools/cmake/helpers/sources.cmake)
 
-set(CH_SRC_DIR ${COMPONENTS_DIR}/connection_handler/src)
+find_package(SDLOpenSSL REQUIRED)
 
-include_directories (
-    ${COMPONENTS_DIR}/connection_handler/include
-    ${COMPONENTS_DIR}/protocol_handler/include/
-    ${COMPONENTS_DIR}/config_profile/include/
-    ${COMPONENTS_DIR}/utils/include/
-    ${ENCRYPTION_INCLUDE_DIRECTORY}
-    ${OPENSSL_INCLUDE_DIRECTORY}
-    ${LOG4CXX_INCLUDE_DIRECTORY}
+include_directories(
+  ${COMPONENTS_DIR}/include
+  ${COMPONENTS_DIR}/connection_handler/include
+  ${COMPONENTS_DIR}/protocol_handler/include
+  ${COMPONENTS_DIR}/config_profile/include
+  ${COMPONENTS_DIR}/utils/include
+  ${COMPONENTS_DIR}/resumption/include
+  ${OPENSSL_INCLUDE_DIRECTORY}
+  ${ENCRYPTION_INCLUDE_DIRECTORY}
 )
 
-set (SOURCES
-    ${CH_SRC_DIR}/connection_handler_impl.cc
-    ${CH_SRC_DIR}/connection.cc
-    ${CH_SRC_DIR}/device.cc
-    ${CH_SRC_DIR}/heartbeat_monitor.cc
+set(PATHS
+  ${CMAKE_CURRENT_SOURCE_DIR}/include
+  ${CMAKE_CURRENT_SOURCE_DIR}/src
 )
 
 set(LIBRARIES
-    ConfigProfile
-    ProtocolLibrary
+  config_profile
+  protocol
+  encryption
+  ${OPENSSL_LIBRARIES}
+  ${ENCRYPTION_LIBRARIES}
 )
```

Comment:
@anosach-luxoft 
Shouldn't we remove 'encryption' because of ${ENCRYPTION_LIBRARIES}?